web/src/api/hooks/tasks.ts

```diff
@@ -280,25 +280,37 @@ export const useGetValidatedPages: QueryHookType<UseTaskForValidatedPagesParams,
             }
         });
 
-export const useSetTaskFinished = (
-    taskId: number,
+interface TaskOptions {
+    taskId: number;
     options?: {
         option_invalid?: string;
         option_edited?: string;
-    }
-) => {
+    };
+}
+
+const finishTask = async ({ taskId, options }: TaskOptions): Promise<TaskModel> => {
     const body = options
         ? {
               annotation_user_for_failed_pages: options.option_invalid || null,
               validation_user_for_reannotated_pages: options.option_edited || null
           }
         : {};
-    return useBadgerFetch({
+    return await useBadgerFetch<TaskModel>({
         url: `${namespace}/tasks/${taskId}/finish`,
         method: 'post'
     })(JSON.stringify(body));
 };
 
+export const useSetTaskFinishedMutation: MutationHookType<TaskOptions, TaskModel> = () => {
```

Comment:
maybe it would be better to make this function clean

Reply:
What do you mean by this?

Reply:
answered in Teams